client/reader/stream/index.jsx

```diff
@@ -507,6 +454,17 @@ export default class ReaderStream extends React.Component {
 		/>;
 	}
 
+	// input fn must only take a single object-like argument.
+	// output is a function that is identical to the input fn except that its option
+	// argument will also have postKey and store, and index in it
+	enhanceWithPostSelectionMetadata = ( fn, postKey, store ) => args =>
```

Comment:
I think my comment oversimplified the alternative.  Its still going to need to be a curried function like 
```javascript
showPost = ( postKey, store ) => args => showSelectedPost( {
  ...args,
  postKey,
  store,
  index: this.state.posts.indexOf( postKey ),
} );
```

For now I think the only reason thats necessary is to differentiate between comment clicks + regular clicks.  So it could potentially be (with some refactoring):

```javascript
showPost = ( postKey, store ) => goToComments => showSelectedPost( {
  goToComments,
  postKey,
  store,
  index: this.state.posts.indexOf( postKey ),
} );
```